app/controllers/RegistrationController.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import controllers.actions.{DataRequiredAction, DataRetrievalAction, IdentifierAction}
+import play.api.mvc.{Action, AnyContent}
+import play.api.mvc.Results.{Ok, Redirect}
+import services.DigitalServicesTaxService
+
+import javax.inject.Inject
+
+class RegistrationController @Inject() (
```

Comment:
I only had a quick look, so I may be wrong. the `RegistrationController` is probably redundant. it is only called/redirected from the `CheckYourAnswersController`, and its logic could be implemented directly within the `CheckYourAnswersController`

Reply:
I've set it up because it is going to be needed to render a page/handle errors at the end of the journey